integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -533,18 +533,20 @@ def fetch_pivot_values(spark):
         "PivotFirst",
         conf=_nans_float_conf)
 
-@approximate_float
-@ignore_order(local=True)
-@incompat
-@pytest.mark.parametrize('data_gen', _init_list_no_nans, ids=idfn)
-@pytest.mark.parametrize('conf', get_params(_confs_with_nans, params_markers_for_confs_nans), ids=idfn)
-def test_hash_reduction_pivot_without_nans(data_gen, conf):
-    assert_gpu_and_cpu_are_equal_collect(
-        lambda spark: gen_df(spark, data_gen, length=100)
-            .groupby()
-            .pivot('b')
-            .agg(f.sum('c')),
-        conf=conf)
+# https://github.com/NVIDIA/spark-rapids/issues/4514
```

Comment:
It would be better to skip the test rather than comment it out:

```python
@pytest.mark.skip(reason='https://github.com/NVIDIA/spark-rapids/issues/4514')
```

Reply:
`xfail` seems appropriate here, as this is code we fully expect to pass but is failing due to known issues.  `skip` should be used for tests that are never expected to pass if attempted, even after bugfixes (e.g.: tests for a feature that a particular Spark version does not support, etc.)

Reply:
It doesn't seem to be failing in github CI, so we probably want to skip?

Reply:
xfail is what we want.  xfail is fine if the test actually passes (it reports it as an XPASS result).

Reply:
There is a config for that I wish we had done a good enough job with things that it would fail. But that is not how it is right now :(